src/App.tsx

```diff
@@ -1,61 +1,142 @@
+import React, { ChangeEvent, FormEvent, useState } from 'react';
+import fetchedUsers from './api/users';
+import fetchedTodos from './api/todos';
+import { User } from './Types';
+import { TodoList } from './components/TodoList';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const todosWithAssociatedUsers = fetchedTodos.map(todo => ({
+  ...todo,
+  user: fetchedUsers.find(user => user.id === todo.userId),
+}));
 
 export const App = () => {
+  const [todoItems, setTodoItems] = useState(todosWithAssociatedUsers);
+  const [formValues, setFormValues] = useState({ title: '', user: '' });
+  const [isValidTitle, setIsValidTitle] = useState(false);
+  const [isValidUser, setIsValidUser] = useState(false);
+  const [displayTitleError, setDisplayTitleError] = useState(false);
+  const [displayUserError, setDisplayUserError] = useState(false);
+  // const [nextId, setNextId] = useState(todoItems.length + 1);
+
+  const handleTitleInputChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value.trimStart();
+
+    setDisplayTitleError(!!value);
+    setIsValidTitle(!!value);
+
+    setFormValues(prevValues => ({
+      ...prevValues,
+      title: value,
+    }));
+  };
+
+  const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const { value } = event.target;
+
+    setDisplayUserError(!!value);
+    setIsValidUser(!!value);
+
+    setFormValues(prevValues => ({
+      ...prevValues,
+      user: value,
+    }));
+  };
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+
+    if (!isValidTitle || !isValidUser) {
+      if (!isValidTitle) {
+        setDisplayTitleError(true);
+      }
+
+      if (!isValidUser) {
+        setDisplayUserError(true);
+      }
+
+      return;
+    }
+
+    const selectedUser = fetchedUsers.find(
```

Comment:
What if we have a few users with exact same name? 
I believe, it's better to use userId here instead of name. This will require you to do some refactoring with selector options and form values state - but this will make your code more reliable :)